src/Chainweb/Pact/Service/PactInProcApi.hs

```diff
@@ -89,22 +90,29 @@ withPactService'
     -> Maybe FilePath
     -> Maybe NodeId
     -> Bool
-    -> (TQueue RequestMsg -> IO a)
+    -> (PactQueue -> IO a)
     -> IO a
-withPactService' ver cid logger memPoolAccess cdbv bhDb pdb dbDir nodeid
-                 resetDb action =
-    mask $ \rst -> do
-        reqQ <- atomically (newTQueue :: STM (TQueue RequestMsg))
-        a <- withLink $
-             PS.initPactService ver cid logger reqQ memPoolAccess cdbv bhDb
-                                pdb dbDir nodeid resetDb
-        evaluate =<< rst (action reqQ) `finally` closeQueue reqQ `finally` wait a
+withPactService' ver cid logger memPoolAccess cdbv bhDb pdb dbDir nodeid resetDb action = do
+    reqQ <- atomically $ newTBQueue pactQueueSize
+    race (server reqQ) (client reqQ) >>= \case
+        Left () -> error "pact service terminated unexpectedly"
+        Right a -> return a
+  where
+    client reqQ = action reqQ
+    server reqQ = runForever logg "pact-service" $
```

Comment:
Is `withPactService'` expected to run forever too?

Reply:
No, only the server component is expect to stay alive as long as the client is running.

In practice, in a production chainweb node, the client above will also run forever, but that's not a concern of this function.